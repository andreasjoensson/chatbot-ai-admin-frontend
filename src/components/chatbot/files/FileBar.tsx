const FileBar = ({ savedFiles, savedUrls }: any) => {
  return (
    <div className="w-full bg-black p-4 text-white">
      <div className="container mx-auto">
        <h2 className="text-lg font-semibold mb-2 text-white">Saved Files</h2>
        <div className="flex flex-wrap">
          {savedFiles.map((file: File) => (
            <div
              key={file.name}
              className="bg-blue-500 text-white p-2 m-2 rounded-md"
            >
              {file.name} Type: {file.type}
            </div>
          ))}
        </div>
        <h2 className="text-lg font-semibold mb-2 mt-4 text-white">
          Saved URLs
        </h2>
        <div className="flex flex-wrap">
          {savedUrls.map((url: string, index: number) => (
            <div
              key={index}
              className="bg-green-500 text-white p-2 m-2 rounded-md"
            >
              {url}
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default FileBar;
