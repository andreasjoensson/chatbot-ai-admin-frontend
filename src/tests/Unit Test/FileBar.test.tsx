import { render } from '@testing-library/react';
import FileBar from '../../components/chatbot/files/FileBar';
import '@testing-library/jest-dom';

describe('FileBar', () => {
  const defaultProps = {
    savedFiles: [
      new File(['file content'], 'file1.txt', { type: 'text/plain' }),
      new File(['file content'], 'file2.txt', { type: 'text/plain' }),
    ],
    savedUrls: ['https://example.com', 'https://example.org'],
  };

  it('should render without errors', () => {
    const { container } = render(<FileBar {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should display saved files with names and types', () => {
    const { getByText } = render(<FileBar {...defaultProps} />);
    expect(getByText('file1.txt Type: text/plain')).toBeInTheDocument();
    expect(getByText('file2.txt Type: text/plain')).toBeInTheDocument();
  });

  it('should display saved URLs', () => {
    const { getByText } = render(<FileBar {...defaultProps} />);
    expect(getByText('https://example.com')).toBeInTheDocument();
    expect(getByText('https://example.org')).toBeInTheDocument();
  });
});
