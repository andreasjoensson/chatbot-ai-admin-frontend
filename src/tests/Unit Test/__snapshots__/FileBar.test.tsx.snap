// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileBar should render without errors 1`] = `
<div>
  <div
    class="w-full bg-black p-4 text-white"
  >
    <div
      class="container mx-auto"
    >
      <h2
        class="text-lg font-semibold mb-2 text-white"
      >
        Saved Files
      </h2>
      <div
        class="flex flex-wrap"
      >
        <div
          class="bg-blue-500 text-white p-2 m-2 rounded-md"
        >
          file1.txt
           Type: 
          text/plain
        </div>
        <div
          class="bg-blue-500 text-white p-2 m-2 rounded-md"
        >
          file2.txt
           Type: 
          text/plain
        </div>
      </div>
      <h2
        class="text-lg font-semibold mb-2 mt-4 text-white"
      >
        Saved URLs
      </h2>
      <div
        class="flex flex-wrap"
      >
        <div
          class="bg-green-500 text-white p-2 m-2 rounded-md"
        >
          https://example.com
        </div>
        <div
          class="bg-green-500 text-white p-2 m-2 rounded-md"
        >
          https://example.org
        </div>
      </div>
    </div>
  </div>
</div>
`;
